sklearn/preprocessing/label.py

```diff
@@ -41,7 +41,12 @@ def _encode_numpy(values, uniques=None, encode=False):
     # only used in _encode below, see docstring there for details
     if uniques is None:
         if encode:
-            uniques, encoded = np.unique(values, return_inverse=True)
+            try:
+                uniques, encoded = np.unique(values, return_inverse=True)
+            except TypeError:
+                 print("Your data contains two different types of values,"
+                       "strings and integers, which are not able to"
+                       "be compared by sorting utils for Pandas.")
```

Comment:
You need to put this in the `_encode_python` one, as unsortable data will be object dtype and end up taking that path, not the numpy version.

Also, instead of printing the message, we would like to raise it again as an error (`raise TypeError("...")`)